packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -769,6 +770,33 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 			0x56e /* summary upload manager should have been initialized */,
 		);
 		const id = await this.odspSummaryUploadManager.writeSummaryTree(summary, context);
+		const { pendingRename } = this.odspResolvedUrl;
+		if (
+			pendingRename !== undefined &&
+			this.config.getBoolean("Fluid.Driver.Odsp.disablePendingRename") !== false
+		) {
+			// This is a temporary file, so we need to rename it to remove the .tmp extension
+			// This should only happen for the initial summary upload for a new file
+			assert(
+				context.ackHandle === undefined &&
```

Comment:
I am assuming this will always run on the first client creating the empty file. If the client dies just after creating an empty file before it makes the first summary, then I think that file remains useless even today?